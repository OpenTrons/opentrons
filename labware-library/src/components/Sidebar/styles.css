/* scoped styles for Sidebar */
@import '@opentrons/components';
@import '../../styles/breakpoints.css';
@import '../../styles/spacing.css';

.sidebar {
  width: 100%;
  padding: var(--spacing-7) var(--spacing-3) 0 var(--spacing-3);
}

.labware_guide {
  border-bottom: var(--bd-light);
  padding-bottom: var(--spacing-7);
}

.labware_guide_container {
  border: var(--bd-light);
  border-radius: 3px;
  padding: var(--spacing-4) var(--spacing-6);
}

.labware_guide_title {
  @apply --font-body-2-dark;

  display: flex;
  align-items: center;
  font-weight: var(--fw-semibold);
}

.labware_guide_icon {
  flex: none;
  width: var(--size-1);
  margin-right: var(--spacing-5);
}

.labware_guide_link {
  display: block;
  line-height: var(--lh-title);
  font-size: var(--fs-body-1);
  margin-top: var(--spacing-3);
  color: var(--c-blue);
}

.filter_manufacturer {
  display: block;
  margin: var(--spacing-7) 0;
}

.filter_manufacturer_label {
  text-transform: uppercase;
  padding: 0 0 var(--spacing-3) var(--spacing-3);
  font-size: var(--fs-caption);
  font-weight: var(--fw-semibold);
  color: var(--c-dark-gray);
}

.filter_manufacturer_select {
  @apply --font-body-1-dark;

  line-height: var(--lh-title);

  /* react-select adds background: white on its <input> for some reason */
  & input {
    background-color: inherit;
  }
}

.filter_category {
  display: flex;
  flex-wrap: wrap;
  padding: 0 var(--spacing-3);
}

.filter_category_item {
  width: var(--size-third);
  margin-bottom: var(--spacing-6);
}

.filter_category_link {
  @apply --font-default-dark;
  @apply --transition-color;
  @apply --clickable;

  line-height: var(--lh-title);
  font-weight: var(--fw-semibold);

  &:hover,
  &.selected {
    color: var(--c-highlight);
  }
}

@media (--medium) {
  .sidebar {
    position: fixed;
    width: var(--size-3);
    height: 100%;
    padding-left: var(--spacing-7);
    padding-right: var(--spacing-5);
    overflow-y: auto;
  }

  .filter_category {
    display: block;
    padding: 0;
  }

  .filter_category_item {
    display: block;
    width: 100%;
  }
}

@media (--large) {
  .sidebar {
    width: var(--size-4);
    padding-left: var(--spacing-7);
  }
}
