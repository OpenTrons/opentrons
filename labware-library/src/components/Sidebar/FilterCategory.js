// @flow
// filter labware by category
import * as React from 'react'
import { Link } from 'react-router-dom'
import cx from 'classnames'

import { getAllCategories, buildFiltersUrl } from '../../filters'
import styles from './styles.css'

import type { FilterParams } from '../../types'

// TODO(mc, 2019-03-18): i18n, duplicated in
//   labware-library/src/components/LabwareList/LabwareCard.js
const EN_CATEGORY_LABELS = {
  all: 'All',
  tubeRack: 'Tube Rack',
  tipRack: 'Tip Rack',
  wellPlate: 'Well Plate',
  trough: 'Trough',
  trash: 'Trash',
  other: 'Other',
}

export type FilterCategoryProps = {
  filters: FilterParams,
}

export default function FilterCategory(props: FilterCategoryProps) {
  const { filters } = props
  const categories = getAllCategories()

  return (
    <ul className={styles.filter_category}>
      {categories.map(c => (
        <li key={c} className={styles.filter_category_item}>
          <Link
            to={buildFiltersUrl({ ...filters, category: c })}
            className={cx(styles.filter_category_link, {
              [styles.selected]: c === filters.category,
            })}
          >
            {EN_CATEGORY_LABELS[c]}
          </Link>
        </li>
      ))}
    </ul>
  )
}
