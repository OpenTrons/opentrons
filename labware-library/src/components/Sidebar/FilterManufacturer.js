// @flow
// filter labware by manufacturer
import * as React from 'react'
import { withRouter } from 'react-router-dom'

import { SelectField } from '@opentrons/components'
import { getAllManufacturers, buildFiltersUrl } from '../../filters'
import styles from './styles.css'

import type { ContextRouter } from 'react-router-dom'
import type { FilterParams } from '../../types'

// TODO(mc, 2019-03-13): i18n
const EN_MANUFACTURER = 'manufacturer'

// TODO(mc, 2019-03-18): i18n
const EN_MANUFACTURER_LABELS = {
  all: 'All',
  generic: 'Generic',
}

export type FilterManufacturerProps = {
  ...ContextRouter,
  filters: FilterParams,
}

export function FilterManufacturer(props: FilterManufacturerProps) {
  const { history, filters } = props
  const manufacturers = getAllManufacturers()
  const options = manufacturers.map(value => ({
    value,
    label: EN_MANUFACTURER_LABELS[value] || value,
  }))

  return (
    <label className={styles.filter_manufacturer}>
      <p className={styles.filter_manufacturer_label}>{EN_MANUFACTURER}</p>
      <SelectField
        className={styles.filter_manufacturer_select}
        name="manufacturer"
        value={filters.manufacturer}
        options={options}
        onValueChange={(_, value) => {
          if (value) {
            history.push(buildFiltersUrl({ ...filters, manufacturer: value }))
          }
        }}
      />
    </label>
  )
}

export default withRouter(FilterManufacturer)
