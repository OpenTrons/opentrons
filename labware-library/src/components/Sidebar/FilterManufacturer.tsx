// filter labware by manufacturer
import { getAllManufacturers, buildFiltersUrl } from '../../filters'
import { MANUFACTURER, MANUFACTURER_VALUES } from '../../localization'
import type { FilterParams } from '../../types'
import styles from './styles.css'
import { SelectField } from '@opentrons/components'
import type { SelectOptionOrGroup } from '@opentrons/components'
import * as React from 'react'
import { withRouter } from 'react-router-dom'
import type { RouteComponentProps } from 'react-router-dom'

export interface FilterManufacturerProps extends RouteComponentProps {
  filters: FilterParams
}

export function FilterManufacturerComponent(
  props: FilterManufacturerProps
): JSX.Element {
  const { history, filters } = props
  const manufacturers = getAllManufacturers()
  const options: SelectOptionOrGroup[] = manufacturers.map(value => ({
    value,
    label: MANUFACTURER_VALUES[value] || value,
  }))

  options.push({ options: [{ label: 'Other', value: 'other' }] })

  return (
    <label className={styles.filter_manufacturer}>
      <p className={styles.filter_label}>{MANUFACTURER}</p>
      <SelectField
        className={styles.filter_manufacturer_select}
        name="manufacturer"
        value={filters.manufacturer}
        options={options}
        onValueChange={(_, value) => {
          if (value) {
            history.push(buildFiltersUrl({ ...filters, manufacturer: value }))
          }
        }}
      />
    </label>
  )
}

export const FilterManufacturer = withRouter(FilterManufacturerComponent)
