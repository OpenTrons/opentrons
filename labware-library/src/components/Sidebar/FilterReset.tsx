// reset all filters button
import { buildFiltersUrl, FILTER_OFF } from '../../filters'
import { CLEAR_FILTERS } from '../../localization'
import type { FilterParams } from '../../types'
import styles from './styles.css'
import { Icon } from '@opentrons/components'
import * as React from 'react'
import { Link } from 'react-router-dom'

export interface FilterResetProps {
  filters: FilterParams
}

export function FilterReset(props: FilterResetProps): JSX.Element | null {
  const { filters } = props
  // TODO (ka 2019-3-09):Should this be moved to Sidebar?
  const { manufacturer, category } = filters
  const filtersCleared = manufacturer === FILTER_OFF && category === FILTER_OFF
  if (filtersCleared) return null

  return (
    <Link
      to={buildFiltersUrl({ ...filters, category: 'all', manufacturer: 'all' })}
      className={styles.filter_reset_link}
    >
      <Icon name="close" className={styles.filter_reset_icon} />
      {CLEAR_FILTERS}
    </Link>
  )
}
