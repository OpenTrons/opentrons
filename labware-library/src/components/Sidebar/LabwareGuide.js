// @flow
// labware filters
import * as React from 'react'

import { Icon } from '@opentrons/components'
import styles from './styles.css'

// TODO(mc, 2019-03-14): i18n
const EN_LABWARE_GUIDE = 'Labware Guide'
const EN_HOW_TO_CHOOSE_LABWARE = 'How to choose labware'
const EN_MAKING_CUSTOM_DEFINITIONS = 'Making custom definitions'

const LINKS = [
  { label: EN_HOW_TO_CHOOSE_LABWARE, href: '#' },
  { label: EN_MAKING_CUSTOM_DEFINITIONS, href: '#' },
]

export default function LabwareGuide() {
  return (
    <section className={styles.labware_guide}>
      <div className={styles.labware_guide_container}>
        <h2 className={styles.labware_guide_title}>
          <Icon
            className={styles.labware_guide_icon}
            name="book-open-page-variant"
          />
          {EN_LABWARE_GUIDE}
        </h2>
        <ul className={styles.labware_guide_list}>
          {LINKS.map((a, i) => (
            <li key={i}>
              <a
                className={styles.labware_guide_link}
                href={a.href}
                target="_blank"
                rel="noopener noreferrer"
              >
                {a.label}
              </a>
            </li>
          ))}
        </ul>
      </div>
    </section>
  )
}
