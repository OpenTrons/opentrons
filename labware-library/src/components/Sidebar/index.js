// @flow
// main application sidebar
import * as React from 'react'

import LabwareGuide from './LabwareGuide'
import FilterManufacturer from './FilterManufacturer'
import FilterCategory from './FilterCategory'
import styles from './styles.css'

export default function Sidebar () {
  return (
    <nav className={styles.sidebar}>
      <LabwareGuide />
      <FilterManufacturer />
      <FilterCategory />
    </nav>
  )
}
