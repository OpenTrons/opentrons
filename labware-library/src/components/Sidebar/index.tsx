// main application sidebar
import type { FilterParams } from '../../types'
import { FilterCategory } from './FilterCategory'
import { FilterManufacturer } from './FilterManufacturer'
import { FilterReset } from './FilterReset'
import { LabwareGuide } from './LabwareGuide'
import styles from './styles.css'
import * as React from 'react'

export interface SidebarProps {
  filters: FilterParams
}

export function Sidebar(props: SidebarProps): JSX.Element {
  const { filters } = props

  return (
    <nav className={styles.sidebar}>
      <LabwareGuide />
      <FilterManufacturer filters={filters} />
      <FilterCategory filters={filters} />
      <FilterReset filters={filters} />
    </nav>
  )
}
