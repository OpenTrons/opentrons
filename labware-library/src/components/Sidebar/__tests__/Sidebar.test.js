// @flow
// Sidebar component tests
import * as React from 'react'
import Renderer from 'react-test-renderer'

import Sidebar from '..'

describe('Sidebar', () => {
  test('component renders', () => {
    const tree = Renderer.create(<Sidebar />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders a <nav>', () => {
    const tree = Renderer.create(<Sidebar />)
    // throws if can't find 'nav'
    tree.root.findByType('nav')
  })
})
