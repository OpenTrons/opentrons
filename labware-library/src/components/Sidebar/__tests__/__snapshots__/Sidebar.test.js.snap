// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component renders 1`] = `
<nav
  className="sidebar"
>
  <LabwareGuide />
  <withRouter(FilterManufacturer)
    filters={
      Object {
        "category": "all",
        "manufacturer": "all",
      }
    }
  />
  <FilterCategory
    filters={
      Object {
        "category": "all",
        "manufacturer": "all",
      }
    }
  />
  <FilterReset
    filters={
      Object {
        "category": "all",
        "manufacturer": "all",
      }
    }
  />
</nav>
`;
